### DISCUSSION TITLE
Can I export to a PDF with multiple pages?

### DISCUSSION THREAD

**Discussion by Dan1ve** [dlemstra/magick-wasm#94] [Q&A] [answered]
Hi, 

thanks for creating such a nice library!
I want to combine multiple input images and export them as a single PDF file, so that every input image is a single page of the output PDF. Is this possible? 

I tried various things with `MagickImageCollection` but couldn't get it to work.  (I know that merging multiple **input** PDFs is not possible)

Thanks!

**Comment by dlemstra** [comment]
That should be possible? Can you show what you have tried so far and where it fails? You should add the images to the `MagickImageCollection ` and then call write with the `pdf` format.

**Comment by Dan1ve** [comment]
Thanks for the fast response! I think I am simply struggling to use the collection API properly. 

Here is some code that works partially (it exports a PDF, but with only one page).
The input is a `FileList` from an `<input />` element, which gets converted to `Uint8Array` objects.

``` ts 
async function processImages(files: FileList) {
	const collection = MagickImageCollection.create();

	for (let i = 0; i < files.length; i++) {
		const arrayBuffer = await files[i].arrayBuffer();
		const array = new Uint8Array(arrayBuffer);

                 // Probably wrong?
		collection.read(array);			
	}

	collection.write(
		(output) => downloadBlob(output, 'test.pdf', 'application/octet-stream'),
			MagickFormat.Pdf
	);
}
``` 

I also tried to use `collection.push()` but I can't figure out how to use the callback-based creation of an `ImageMagik` image without running into `Object has been disposed` problems. 

I'm pretty sure there is an obvious solution for this, but I couldn't find a matching example in the  docs / test code.

**Reply by dlemstra** [reply, marked as answer]
I was suspecting that this was happening. And you are correct about the `collection.read(array);` line. You will need to read and an image and add it without disposing it. You can do that with `collection.push(MagickImage.Create(array))`. But I should probably add an extra method where you can just do `collection.newMethod(array)`. But what method name would you expect for this?

**Comment by Dan1ve** [comment]
`collection.push(MagickImage.create(array))` indeed did the trick, thanks a lot! 🎉

Maybe the potential new method could be called `collection.pushImage()`, which aligns a bit with the `Array` API? Another option would be `add()` or `append()` I guess.

Kudos for the API and Typescript definitions, this is really a joy to use. And super cool to have this locally in the browser :rocket:

**Comment by dlemstra** [comment]
`pushImage` was also my first thought. Will think about it and come back here if I decide to use another name.
